File: GFG POTD/022-September-2025/023-queue-reversal.java
// https://www.geeksforgeeks.org/problems/queue-reversal/1/




import java.util.*;

class Solution {
    // TC : O(N)
    // SC : O(N)
    public void reverseQueue(Queue<Integer> q) {
        Stack<Integer> stk = new Stack<>();
        while (!q.isEmpty()) {
            stk.push(q.poll());
        }
        while (!stk.isEmpty()) {
            q.offer(stk.pop());
        }
    }
}

File: GFG POTD/022-September-2025/023-queue-reversal.cpp
// https://www.geeksforgeeks.org/problems/queue-reversal/1/




#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    // TC : O(N)
    // SC : O(N)
    void reverseQueue(queue<int> &q) {
        stack<int> stk;
        while (!q.empty()) {
            stk.push(q.front());
            q.pop();
        }
        while (!stk.empty()) {
            q.push(stk.top());
            stk.pop();
        }
    }
};